### ISSUE TITLE
About the generation results

### ISSUE THREAD

**Issue by muradilkurban** [GradientSpaces/respace#4]
<img width="600" height="460" alt="Image" src="https://github.com/user-attachments/assets/3ca2e57b-cdbd-41c9-85b9-bdf14a60d564" />

Hello! It's me again.
Some of the generated 3D scenes are excellent, but at least five out of ten are similar to this one. Do you know why this is the case?

**Comment by mnbucher** [comment]
for full scene synthesis (which seems to be the case above?), the major bottleneck is that the zero-shot LLM produces a list of object prompts, and mentions "bed" twice. this places two beds into a packed room already, hence the beds overlap heavily. in other cases, the zero-shot LLM includes "correctly" only one bed, but then the SG-LLM would place 2 beds, with one bed matching very poorly the intended object prompt (e.g., it places a bed when prompted with "wardrobe"). could you paste the list of object prompts that the zero-shot LLM produces, and the full SSR of the scene above? usually the exported/final SSR contains both the "query" (short object prompt) and the desc of the sampled asset. this can help to narrow down the cause of the overlap. in general, we observed the same behavior in our experiments, i.e. some very strong overlap or bad object prompt lists. may i ask what you chose for BoN here? (best-of-n?)

**Comment by muradilkurban** [comment]
I do remember "bed" was mentioned twice in "add commands". But even though there is no object of same category, overlapping still occurred. Actually what I am curious to know is why there is so much overlap. When I generated fewer objects, the overlapping issue improved significantly.  It feels like the cause of the overlap issue is that the floor area isn't large enough to fit seven or eight objects. By the way, I replaced the zero-shot LLM with Qwen/Qwen2.5-7B-Instruct because I didn’t get permission to use meta-llama/Meta-Llama-3.1-8B-Instruct. Do you think it will make a significant difference in the generation results?  I didn't change the BoN value—it should be the default vaule.

**Comment by mnbucher** [comment]
yes, overlapping objects is an issue that we observed in many scenes. our method has not solved scene synthesis unfortunately. one main motivation was to reduce overlapping via GRPO / preference alignment, but since we had such a low learning rate, the improvements are only directional. usually, the BoN=8 reduces overlaps significantly, since we can choose the best placement. you should double-check what setting you use here. do you use the eval shell scripts provided by us? see here:

https://github.com/GradientSpaces/respace/blob/46cdbed01a83ee0baacba5c422eed9cac6b6de8d/src/scripts/eval/respace_eval_full.sh#L5

if you have a good hypothesis how to resolve the overlapping behaviour completely, i would be very curious.

my core conclusion was that in most cases, the zero-shot LLM produces too many object prompts, and the space is too small. hence, SG-LLM has no choice but to place it somewhere, causing the overlap. this aligns with your observations. since the zero-shot LLM is "uninformed" about the space constraints, this causes a bit of a bottleneck. we have acknowledged this issue in our limitations in the paper. there are various ways to build around that, but we leave it for future work.

i have not tried GPT-5 on scene synthesis yet. might be interesting to see how LayoutGPT, with GPT-5, performs on the same benchmarks, since it used GPT-4 previously.

another idea would be to "replace" SG-LLM with GPT-5 as a baseline (i.e., given the existing SSR and an object prompt, GPT-5 tries to predict the next object, instead of SG-LLM). i recognize this could have been an interesting baseline as well to quantify how much fine-tuning (SG-LLM) actually helps.

regarding your question on Qwen2.5-7B-Instruct vs. Meta-Llama-3.1-8B-Instruct, this is a good question. i would not expect too many differences, since qwen usually performs stronger than llama. however, the prompts might need some adjustment. i am not sure how much the performance will differ.

**Comment by muradilkurban** [comment]
Okay. For the next steps, I could perhaps run this script `respace/src/scripts/eval/respace_eval_full.sh`, or it would also be interesting to compare the improvement of SG-LLM after fine-tuning versus Qwen2.5-1.5B-Instruct on the 3D scene synthesis task. In any case, thank you very much for your efforts in advancing the field of 3D scene synthesis.

**Comment by mnbucher** [comment]
thanks for checking out our work so in-depth, i really appreciate it! and yes, more experiments are always welcome. if you encounter any issues with the results, let me know, and we can open a new issue. it seems that the overlapping is indeed caused by e.g., two prompts containing "bed", for the example screenshot above.

**Comment by mnbucher** [comment]
hi @muradilkurban, just wanted to say thanks again for checking out our work. i have a few ideas for future work and how to possibly improve generation results during inference. don't hesitate to reach out via email and we can set up a chat. of course, if you are using our work as a baseline for your own method, that is also very nice.

**Comment by muradilkurban** [comment]
ok! thank u very much too.
